pymodbus/repl/server/main.py

```diff
@@ -126,7 +128,7 @@ def run(
         "--modbus-port",
         "-p",
         help='Modbus port'),
-    modbus_unit_id: List[int] = typer.Option(
+    modbus_unit_id: list[int] = typer.Option(
```

Comment:
Let's use explicit annotations from `typing` module , `list[int]` might result in `TypeError: 'type' object is not subscriptable`

Reply:
List from typing is deprecated in 3.9, and the suggestion is to use list (builtin type).

The error you get is because I forgot an import at the top "from __future__ import annotations"

So let us not use List from the typing module, or we no longer support higher versions.